README.md

```diff
@@ -13,124 +25,292 @@ The prime purpose of the reader is to transform lab-defined data into community-
+- To update (if needed) the config file, a set of rules needs to be followed:
+  - The dictionary in config files have the following meaning:
     ```
     "/ENTRY[entry]/INSTRUMENT[instrument]/lock_in/harmonic_order_N": {"D1": {"value": "/Lock-in/Harmonic D1/value"},
                                                                       "D2": {"value": "/Lock-in/Harmonic D2/value"}},
     ```
-    Here, the `N` in field `harmonic_order_N`, can be considered as the name of dimensions, can be replaced by `D1` and `D2` to  write two `harmonic_order`.
+    Here, the part `N` in field `harmonic_order_N`, can be considered as the name of dimensions, can be replaced by `D1` and `D2` to  write two `harmonic_order` and can be extended for further dimensions.
   - List for the same concept
     ```
     "/ENTRY[entry]/INSTRUMENT[instrument]/piezo_config/active_calib": ["/Piezo Configuration/Active Calib.",
                                                                        "/Piezo Calibration/Active Calib."],
     ```
-    For different type of software versions the raw data path could be different for the same
-    concept. For example, Nanonis software `generic 5e` has `/Piezo Configuration/Active Calib.`
-    and generic 4.5 has `/Piezo Calibration/Active Calib.` for the same concept `/ENTRY[entry]/INSTRUMENT[instrument]/piezo_config/active_calib`.
+    For different type of software versions the raw data path could be different for the same concept. For example, Nanonis software `generic 5e` has `/Piezo Configuration/Active Calib.` and generic 4.5 has `/Piezo Calibration/Active Calib.` for the same concept `/ENTRY[entry]/INSTRUMENT[instrument]/piezo_config/active_calib`.
+  - In the config file the concept that takes data from ELN is denoted by `@eln` otherwise data will come from experimental raw files.
+  - Importantly one should note the `NXdata` concept `/ENTRY[entry]/DATA[data]` it takes dict of the list. Each key (`0`, `1` ...) of the dict refers to a NXdata group of data fields `bias` and `current` for multiple given setup, saying, with and without `filter` check points. For other setup, one can extend the dict following the same convenstion used here.
+
+## Run the Reader from CLI
+In command line cli input files come as positional arguments and the other comes a keyword argument.
+### Run STS Reader
+The following command can be used to run the `STS` reader:
+```bash
+!dataconverter \
+--reader sts \
+--nxdl NXsts \
+$<path-to STS_nanonis_generic_5e_1.dat> \
+$<path-to config_file_for_dat.json> \
+$<path-to Nanonis_Eln.yaml> \
+--output ./final_sts_dev.nxs
+```
+
+### Run STM Reader
+Use the following command to run the `STM` reader
+```bash
+!dataconverter \
+--reader sts \
+--nxdl NXsts \
+$<path-to STM_nanonis_generic_5e.sxm> \
+$<path-to config_file_for_sxm.json> \
+$<path-to Nanonis_Eln.yaml> \
+--output final_stm_dev_.nxs
+```
+## Want to Contribute or Extend the Reader Functionality
```

Comment:
You have this point now twice, once as `## Want to Contribute or Extend the Reader Functionality` and once as `### How to contribute / How to extend the reader functionality` (line 284 ff.). Please consider merging.

Reply:
Noted!